nomos-libp2p/src/lib.rs

```diff
@@ -51,14 +62,57 @@ impl Behaviour {
                 .max_transmit_size(DATA_LIMIT)
                 .build()?,
         )?;
-        Ok(Self { gossipsub })
+
+        let store = kad::store::MemoryStore::new(peer_id);
+        let mut kad_config = kad::Config::new(StreamProtocol::new(KAD_PROTOCOL_NAME));
+        kad_config.set_periodic_bootstrap_interval(Some(Duration::from_secs(10))); // this interval is only for proof of concept and should be configurable
+        let kademlia = kad::Behaviour::with_config(peer_id, store, kad_config);
+
```

Comment:
All these we can make configurable. Even the whole kademlia can be optional, i order to cancel it if needed for testing, etc...

Reply:
At the moment it doesn't seem straightforward to make kademlia optional, maybe you have an idea in mind?

Reply:
I think that derive network behavior works for `Option<T>` params, but I am not 100% sure. I will confirm and try to find the new way if not

Reply:
Ah ok, just tried it, maybe not, but I guess a I can always wrap behavior and delegate only the methods for the Some fields. There must be a way :)

Reply:
Nice, that should work and could be toggled at runtime! So Identify and Kademlia would both be optional?

Reply:
We should make both optional. We also need to see if we need mdns maybe instead of identify, but I guess identify makes more sense

Reply:
Relevant doc: https://docs.rs/libp2p-kad/latest/libp2p_kad/. I have see  a few impla with mdns instead

Reply:
they could be composed. For example we could (and I think we should because they serve the same purpose) wrap kademlia+identify in its own behaviour (lets call it `NomosDiscovery`, this can later be expanded for adding other discovery protocols for example) then rewrap with toggle.

Reply:
I am not 100% sure, but I think that identify might be able to help with NAT as well, so we could, in theory, use it without kademlia. It is useful for getting observed address and other addr information upon connection.

Reply:
Agree, let's use it.